### ISSUE TITLE
sspinegotiate with https transport

### ISSUE THREAD

**Issue by maclennann** [chef/winrm-s#19]
I'm trying to use `winrm-s` with `sspinegotiate` authentication over an `HTTPS` transport and I'm having some trouble.

Using the HTTP transport works fine, but when I change to HTTPS and port 5986, I get back a 401.

I'm able to connect using HTTPS through powershell with: `C:\> enter-pssession computer.example.com -authentication negotiate -usessl -credential example\username`.

However, the following script fails with a 401 (but succeeds when the endpoint is http/5985):
```ruby
require 'bundler/setup'
require 'winrm-s'

endpoint = "https://computer.example.com:5986/wsman"
winrm = WinRM::WinRMWebService.new(endpoint, :sspinegotiate,
  :user=>"example\username",
  :pass=>"password",
  :host=>"computer.example.com")

winrm.cmd('ipconfig /all') do |stdout, stderr|
  STDOUT.print stdout
  STDERR.print stderr
end
```

I turned on `debug_dev` in the httpclient gem and it looks like it's getting the challenge and trying to authenticate, but is never successfully able to. I can throw the output up in a gist or something if anyone wants to see it.

This is with winrm-s 0.2.2, winrm 1.2.0, and I've tried both httpclient 2.5.0 and 2.6.0.1.

I'm going to do a bit more digging, I just wanted to open this early in case I was missing something super-obvious or anyone had any pointers (or if this is a known limitation).

**Comment by chrisbaldauf** [comment]
+1

**Comment by adamedx** [comment]
@maclennann, you may be hitting this issue: https://github.com/chef/knife-windows/issues/176. Otherwise, SSL should work fine -- see "platform authentication support" table at https://github.com/chef/knife-windows/commit/635058b29f708774f539972952bc605db0a17f3c?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8 for what should work. Note that getting it to work in releases prior to knife-windows 1.0.0 (which is not yet released) requires using a name in the format 'domain\user' where negotiate is the default in 1.0.0.

**Comment by maclennann** [comment]
Ah-ha! That specific issue wasn't the problem, but thanks for the pointers.

I looked at how knife implements it and I was able to figure out the problem - I was still trying to use `:sspinegotiate` through `winrm-s`  to talk over SSL because I thought that was the only way to auth domain accounts. I switched back to using the regular `winrm` gem with the `:ssl` transport for this scenario and everything magically worked. Thanks!
